### PR TITLE
[match data] allow sending match data in create transaction API call

### PR DESCRIPTION
**Pull request by nwadams**
allow sending seeded `match_data` to be validated against extracted data in the API. This data will be used to compare against extracted data to verify correctness.

### CODE REVIEW

File: lib/Client.php [hunk 1]
```diff
@@ -68,6 +68,25 @@ private function parseCreateTransactionOptions(array $opts, array $payload): arr
...
+      $matchData = $opts['matchData']
+      $payload['match_data'] = array();
+      if (array_key_exists('givenName', $matchData)) {
+        $payload['match_data']['given_name'] = $matchData['givenName']
+      }
+      if (array_key_exists('familyName', $matchData)) {
+        $payload['match_data']['family_name'] = $matchData['familyName']
+      }
+      if (array_key_exists('middleName', $matchData)) {
+        $payload['match_data']['middel_name'] = $matchData['middleName']
```
Comment (andrewsli) [inline lib/Client.php]: `middle_name`

**Review by andrewsli — APPROVED** [review]
just one typo

**Comment by benbeltzer** [comment]
Just to confirm, we're planning to _not_ update the docs until this is more widely available, correct?

**Comment by nwadams** [comment]
@benbeltzer after this lands I plan to push a 2.1.0 beta release with the feature and a few docs included. Wanted to land the feature first.

**Pull request merged**